convert2rhel/utils.py

```diff
@@ -702,44 +704,51 @@ def is_dir_empty(path):
+    Replace secret values passed through a command line with asterisks.
 
-    This function takes a list of arguments which will be passed
-    in a transformation process where we will replace any secret values
-    with an fixed size of asterisks (*) and returns a new list containing
-     the arguments with this transformation.
+    This function takes a list of command line arguments and returns a new list containing where any secret value is
+    replaced with a fixed size of asterisks (*).
 
-    :arg args: An argument list which may contain secret values.
+    Terminology:
+     - an argument is one of whitespace delimited strings of an executed cli command
+       Example: `ls -a -w 10 dir` ... four arguments (-a, -w, 10, dir)
+     - an option is a subset of arguments that start with - or -- and modify the behavior of a cli command
+       Example: `ls -a -w 10 dir` ... two options (-a, -w)
+     - an option parameter is the value following an aption if the option requires a value
+       Example: `ls -a -w 10 dir` ... one option parameter (10)
+
+    :arg args: A list of command line arguments which may contain secret values.
+    :arg secret_options: A set of command line options requiring sensitive or private information.
     :returns: A new list of arguments with secret values hidden.
     """
-    obfuscation_string = "*" * 5
-
     sanitized_list = []
     hide_next = False
     for arg in args:
         if hide_next:
-            # Second part of a two part secret argument (like --password *SECRET*)
-            arg = obfuscation_string
+            # This is a parameter of a secret option
+            arg = OBFUSCATION_STRING
             hide_next = False
 
-        elif arg in secret_args:
-            # First part of a two part secret argument (like *--password* SECRET)
+        elif arg in secret_options:
+            # This is a secret option => mark its parameter (the following argument) to be obfuscated
             hide_next = True
 
         else:
-            # A secret argument in one part (like --password=SECRET)
-            for problem_arg in secret_args:
-                if arg.startswith(problem_arg + "="):
-                    arg = "{0}={1}".format(problem_arg, obfuscation_string)
+            # Handle the case where the secret option and its parameter are both in one argument ("--password=SECRET")
+            for option in secret_options:
+                if arg.startswith(option + "="):
+                    arg = "{0}={1}".format(option, OBFUSCATION_STRING)
 
         sanitized_list.append(arg)
 
     if hide_next:
         loggerinst.debug(
-            "Passed arguments had unexpected secret argument,"
-            " '{0}', without a secret".format(sanitized_list[-1])  # lgtm[py/clear-text-logging-sensitive-data]
+            "Passed arguments had an option, '{0}', without an expected"
+            " secret parameter".format(sanitized_list[-1])  # lgtm[py/clear-text-logging-sensitive-data]
```

Comment:
LGTM bot is gone so can remove this
```suggestion
            " secret parameter".format(sanitized_list[-1])
```

Reply:
That change was not applied, @SpyTec do you want to proceed on merging this and removing the comment later?

Reply:
Oh, I must have missed that. I can fix it, no problem. We still have a couple other PRs left to finish, so we don't need to merge this now.